frontend/src/routes/Credentials/CredentialsForm.test.tsx

```diff
@@ -444,4 +446,70 @@ describe('add credentials page', () => {
     await clickByText('Next')
     await waitForText('This is a required field.', true)
   })
+
+  it('should create kubevirt (Red Hat Virtualization) credentials without external infra', async () => {
+    render(<Component credentialsType={Provider.kubevirt} />)
+    const providerConnection = createProviderConnection('kubevirt', {
+      pullSecret: '{"pull":"secret"}\n',
+      'ssh-publickey': 'ssh-rsa AAAAB1 fakeemail@redhat.com\n',
+    })
+
+    // Render the form and fill in the fields
+    await typeByTestId('credentialsName', providerConnection.metadata.name!)
+    await selectByText('Select a namespace for the credential', providerConnection.metadata.namespace!)
+    await clickByText('Next')
+
+    // (no external infra)
+    await clickByText('Next')
+
+    // Fill in the pull secret and SSH public key
+    await typeByTestId('pullSecret', providerConnection.stringData?.pullSecret!)
+    await typeByTestId('ssh-publickey', providerConnection.stringData?.['ssh-publickey']!)
+    await clickByText('Next')
+
+    // Add Credentials
+    const createNock = nockCreate({ ...providerConnection })
+    await clickByText('Add')
+    await waitForNock(createNock)
+  })
+
+  it('should create kubevirt (Red Hat Virtualization) credentials with external infra', async () => {
+    render(<Component credentialsType={Provider.kubevirt} />)
+
+    // Define the provider connection data
+    const providerConnection = createProviderConnection('kubevirt', {
+      pullSecret: '{"pull":"secret"}\n',
+      'ssh-publickey': 'ssh-rsa AAAAB1 fakeemail@redhat.com\n',
+      kubeconfig: 'kubeconfig',
+      externalInfraNamespace: 'external-namespace',
+    })
+
+    // Render the form and fill in the fields
+    await typeByTestId('credentialsName', providerConnection.metadata.name!)
+    await selectByText('Select a namespace for the credential', providerConnection.metadata.namespace!)
+    await clickByText('Next')
+
+    // Click on the external infrastructure checkbox
+    await waitFor(() =>
+      expect(
+        screen.getByRole('heading', { level: 2, name: /Select to enable and store external infrastructure details/i })
+      ).toBeInTheDocument()
+    )
+    await clickByTestId('isExternalInfra')
```

Comment:
Better to click by what the user sees. (Text or by roles + text.) I think we'll be able to change to text once we reduce the duplication in labels.